### PR TITLE
Inspect use of ordereddict

### LINKED ISSUE #1330: Explore current uses of OrderedDict, and whether these can use Dict in Python 3.7+
We started using OrderedDict in some situations in earlier (minimum) versions of python that we supported, since they preserved ordering.

Since we now depend upon Python 3.7+ (in new releases onwards) then we can reconsider whether we need these uses, and can instead use a simple dict instead.

https://docs.python.org/3.7/library/collections.html#ordereddict-objects

### PR DESCRIPTION
**Pull request by israelsgalaxy**
**What does this PR do?**  
This PR inspects all uses of OrderedDict in the codebase (except in tests) and replaces it with a Dict when applicable

Fixes #1330 

**Tested?** 
- [x] Manually
- [x] Existing tests (adapted, if necessary)
- [ ] New tests added (for any new behavior)
- [x] Passed linting & tests (each commit)

**Commit flow**
- first commit changed type of ordered_emojis to Dict
- second commit regenerated EMOJI_DATA as a Dict
- third commit changed type of active_emoji_data to Dict
- fourth commit changed type of KEY_BINDINGS to Dict
- fifth commit changed type of HELP_CATEGORIES to Dict
- sixth, seventh, eighth, ninth, thirteenth commits changed type of topic_links and message_links to Dict
- tenth commit changed type of matches to Dict
- eleventh commit changed type of event_actions to Dict
- twelfth commit changed type of autocomplete_map to Dict 

**Notes**
I am working on adding the screenshots from my manual testing

### CODE REVIEW

**Review by theViz343 — COMMENTED** [review]
Thanks for creating the PR! This seems to modify all instances of OrderedDict to Dict without changing the appearance and any functionality:thumbsup:. 
The commit message structure, in general, looks good and the descriptions are detailed. You could add periods at the end of the each commit title to improve punctuation.

File: zulipterminal/model.py [hunk 1]
```diff
@@ -140,25 +140,23 @@ def __init__(self, controller: Any) -> None:
             "zulip_version",
         ]
 
+        # fmt: off
```
[the remark this answers is not in the document]
Comment (theViz343) [reply zulipterminal/model.py:143]: You're right. I looked at black's code formatting style, and it has a rule called the [magic trailing comma](https://github.com/psf/black/blob/main/docs/the_black_code_style/current_style.md#the-magic-trailing-comma), which overrides the line length rule as mentioned [here](https://github.com/psf/black/blob/main/docs/the_black_code_style/current_style.md#line-length). 
Then, I think it's best to keep fmt off as you've done.

**Review by neiljp — COMMENTED** [review]
@israelsgalaxy Thanks for working on this :+1:

The screenshots were helpful, and I also checked myself - all looks good :)

I've just pushed back to the PR with some changes:
- adding an initial commit, since the unicode emoji file was not up to date from the server
- adding more detail of what was changed in each commit title, and adjusting some commit text otherwise
- squashing some similar commits together where it is clear or related

The code diff should be identical, after accounting for rebasing against `main`, though the `range-diff` will provide more details of the commit changes eg. commit text.

I'll likely merge soon.

**Comment by neiljp** [comment]
@israelsgalaxy Thanks again, merging now :tada:

**Pull request merged**